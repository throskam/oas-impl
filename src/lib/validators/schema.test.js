const createSchemaValidator = require('./schema')

describe('Type', () => {
  const schema = {
    type: 'integer'
  }

  const validator = createSchemaValidator(schema)

  it('should return no errors when the value is valid', () => {
    const payload = { value: 1 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is invalid', () => {
    const payload = { value: 'string' }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is missing', () => {
    expect(validator()).toMatchSnapshot()
  })
})

describe('Custom format', () => {
  const schema = {
    type: 'number',
    format: 'even'
  }

  const validator = createSchemaValidator(schema, {
    format: {
      even: {
        validator: {
          type: 'number',
          validate: val => {
            return val % 2 === 0
          }
        }
      }
    }
  })

  it('should return no errors when the value is even', () => {
    const payload = { value: 2 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is odd', () => {
    const payload = { value: 1 }
    expect(validator(payload)).toMatchSnapshot()
  })

  it('should return a validation error when the value is missing', () => {
    expect(validator()).toMatchSnapshot()
  })
})
