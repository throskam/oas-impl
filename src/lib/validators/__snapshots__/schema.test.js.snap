// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom format should return a validation error when the value is missing 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be number",
      "params": Object {
        "type": "number",
      },
      "schemaPath": "#/type",
    },
    "path": "schema",
    "rule": "schema-invalid",
    "value": undefined,
  },
]
`;

exports[`Custom format should return a validation error when the value is odd 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "format",
      "message": "must match format \\"even\\"",
      "params": Object {
        "format": "even",
      },
      "schemaPath": "#/format",
    },
    "path": "schema",
    "rule": "schema-invalid",
    "value": 1,
  },
]
`;

exports[`Custom format should return no errors when the value is even 1`] = `Array []`;

exports[`Type should return a validation error when the value is invalid 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "schema",
    "rule": "schema-invalid",
    "value": "string",
  },
]
`;

exports[`Type should return a validation error when the value is missing 1`] = `
Array [
  Object {
    "ajv": Object {
      "instancePath": "",
      "keyword": "type",
      "message": "must be integer",
      "params": Object {
        "type": "integer",
      },
      "schemaPath": "#/type",
    },
    "path": "schema",
    "rule": "schema-invalid",
    "value": undefined,
  },
]
`;

exports[`Type should return no errors when the value is valid 1`] = `Array []`;
